app/lib/package/backend.dart

```diff
@@ -1007,22 +997,15 @@ class PackageBackend {
             version.package, version.version!);
       }
 
+      // If the package exists, check for authorization
+      if (package != null) {
+        await _checkUploadAuthorization(package!, agent);
+      }
+
       // If the package does not exist, then we create a new package.
       if (package == null) {
         _logger.info('New package uploaded. [new-package-uploaded]');
         package = Package.fromVersion(newVersion);
```

Comment:
I don't think we allow bots to publish the first version!

We probably should consider following this up with a mechanism for publishers to create packages without any versions (but that seems like a follow up).

Reply:
Bots won't be able to publish, because there won't be any `AutomatedPublishing` object on the non-existing/newly created `Package` entity. Maybe there could be another configuration elsewhere that pre-authorizes the agent to publish a new package, but that is for another day.